sentry-rails/spec/sentry/rails_spec.rb

```diff
@@ -91,6 +91,52 @@
     end
   end
 
+  context "at exit" do
+    before do
+      make_basic_app
+      Rails.application.load_runner
+    end
+
+    def capture_in_separate_process(exit_code:)
+      pipe_in, pipe_out = IO.pipe
+
+      fork do
+        pipe_in.close
+
+        allow(Sentry::Rails).to receive(:capture_exception) do |event|
+          # raise "hello"
+          pipe_out.puts event
+        end
+
+        # silence process
+        $stderr.reopen('/dev/null', 'w')
+        $stdout.reopen('/dev/null', 'w')
+
+        exit exit_code
+      end
+
+      pipe_out.close
```

Comment:
@st0012 do you have opinions on this test, seems a bit like too much extra work

Reply:
Note this same test pattern was [copied from sentry-raven](https://github.com/getsentry/sentry-ruby/blob/94c28fe9a7fd1c15a10eeeb0e1c4fcac099d4166/sentry-raven/spec/raven/raven_spec.rb#L17-L41)